### DISCUSSION TITLE
ePub rendering (newer versions)

### DISCUSSION THREAD

**Discussion by pstrg** [sumatrapdfreader/sumatrapdf#3987] [Q&A]
3.3 had an issue rendering ePub files with line numbers (usual in poetry): they showed up together with the text, like this:
![sumatrapdf 3 3 3](https://github.com/sumatrapdfreader/sumatrapdf/assets/13485213/f60d2ee3-4368-449a-a3b9-07fe4f8f9c63)

It's way better now, line numbers are shown detached, as they should and titles are shown with different size.
In fact, there should not be a line break after each line number, but that's not so problematic.
However, I could not find in Advanced Settings how to
- avoid line breaking;
- choose left align instead of justify;
is there a way to adjust those?
![sumatrapdf 3 5 2](https://github.com/sumatrapdfreader/sumatrapdf/assets/13485213/8faa84e5-438d-489b-96f5-b50cbeb49e0f)

For reference, this is how Calibre viewer renders the same file:
![calibre](https://github.com/sumatrapdfreader/sumatrapdf/assets/13485213/3c781876-1596-4b4a-b506-824b4cbe17f2)

(Images from a Portuguese translation of Homer's Iliad.)

**Comment by kjk** [comment]
Sadly not.

ePub is just html and there's a trade-off in rendering.

3.3 didn't understand more complicated html but allowed more customizability.

3.4+ understand more html and renders it more accurately but that makes is harder to ignore explicit html instructions (i.e. customize justification or line breaking)
